agent/run_job.go

```diff
@@ -148,6 +157,50 @@ func (r *JobRunner) Run(ctx context.Context) error {
 	return nil
 }
 
+func (r *JobRunner) prependTimestampForLogs(s string, args ...any) []byte {
+	switch {
+	case r.conf.AgentConfiguration.ANSITimestamps:
+		return []byte(fmt.Sprintf(
+			"\x1b_bk;t=%d\x07%s",
+			time.Now().UnixNano()/int64(time.Millisecond),
+			fmt.Sprintf(s, args...),
+		))
+	case r.conf.AgentConfiguration.TimestampLines:
+		return []byte(fmt.Sprintf(
+			"[%s] %s",
+			time.Now().UTC().Format(time.RFC3339),
+			fmt.Sprintf(s, args...),
+		))
+	default:
+		return []byte(fmt.Sprintf(s, args...))
+	}
+}
+
+func (r *JobRunner) verificationFailureLogs(behavior string, err error) {
```

Comment:
for readability in this func it might be nice to add something like 
```Go
jobLog := func(s string, args ...any) {
  l := fmt.Sprintf(s, args)
  r.logStreamer.Process([]byte(r.prependTimestampForLogs(l)))
}
```
and then use it like

```Go
jobLog("%s Job verification failed\n", prefix)
```